lua/nvim-treesitter/query.lua

```diff
@@ -7,10 +7,30 @@ local function read_query_file(fname)
   return table.concat(vim.fn.readfile(fname), '\n')
 end
 
+-- Some treesitter grammars extend others.
+-- We can use that to import the queries of the base language
+M.base_language_map = {
+  cpp = { 'c' },
+  tsx = { 'js' },
+}
```

Comment:
could you only make key to strings mapping ? Or could a language have multiple base languages ?

Reply:
I though this would be a more general solution if the situation of multiple base languages ever arrises. If a user creates their own mash-up language parser they can import multiple queries.

But in general this problem is tackled by embedded languages. The feature is a bit newer so I don't know whether all web-stuff parsers already use it or whether they just import another grammar and create a single AST.

Reply:
Also this is currently the only way to extend extensions. For my fantasy-cpp-parser I could import both `c` and `cpp`.

Reply:
To me a base language is a language from which you extend to other ones. Embedding another language is another issue (C in CPP vs HTML in python is definitely not the same thing).
I don't know how embedding solutions will work though, maybe parser definitions should have associated filetypes like one extended language and potential embedded languages. @vigoux what's you opinion on this matter ?

Reply:
I guess we can forget about the one-language-embedding-multiple-others case.

The only question is whether dependency chains are possible: C extends B and B extends A. Then, C would require queries A and B.

Reply:
Or should we just simplify to pure string until we encounter the problem for the first time?

Reply:
> I guess we can forget about the one-language-embedding-multiple-others case.

Are you sure ? I mean html does have css and javascript but embedding will probably mean different trees, different parsers and so different highlight files.

For now, we should simplify to pure string. But it might be problematic for stuff like javascript in typescript in react. Not sure, but if this is the only case, i think we can forget about multiple highlighting files and create one for this specific purpose.

Reply:
But in the case of injected languages, we will just have complete distinct set of queries. We can not simply concatenate..